lib/folio_sync/archives_space/marc_exporter.rb

```diff
@@ -0,0 +1,79 @@
+module FolioSync
+  module ArchivesSpace
+    class MarcExporter
+      ONE_DAY_IN_SECONDS = 24 * 60 * 60
+      PAGE_SIZE = 200
+
+      def initialize
+        @logger = Logger.new($stdout) # Ensure logger is initialized first
+        @client = FolioSync::ArchivesSpace::Client.instance
+      end
+
+      def export_recent_resources(modified_since = nil)
+        modified_since = modified_since ||= Time.now.utc - (ONE_DAY_IN_SECONDS * 20)
+
+        @client.get_all_repositories.each do |repo|
+          next log_repository_skip(repo) unless repo['publish']
+
+          repo_id = extract_id(repo['uri'])
+          export_resources_from_repository(repo_id, modified_since)
+        end
+      end
+
+      private
+
+      def export_resources_from_repository(repo_id, modified_since)
+        query_params = build_query_params(modified_since)
+
+        @client.retrieve_paginated_resources(repo_id, query_params) do |resources|
+          resources.each do |resource|
+            log_resource_processing(resource)
+            export_marc_for_resource(repo_id, extract_id(resource['uri']), resource['identifier'])
+          end
+        end
+      end
+
+      def export_marc_for_resource(repo_id, resource_id, bib_id)
+        marc_data = @client.fetch_marc_xml_resource(repo_id, resource_id)
```

Comment:
```suggestion
        marc_data = @client.fetch_marc_xml_resource(repo_id, resource_id)
        Rails.logger.error("No MARC found for repo #{repo_id} and resource_id #{resource_id}")
```

Might be good to log an error if this method is ever called and no MARC is available, just so we know to look into it later.